src/targets/__tests__/awsLambda.test.ts

```diff
@@ -0,0 +1,216 @@
+import {
+  BaseArtifactProvider,
+  RemoteArtifact,
+} from '../../artifact_providers/base';
+import { ConfigurationError } from '../../utils/errors';
+import {
+  AwsLambdaTarget,
+  getAwsLayerName,
+  defaultLayerName,
+} from '../awsLambda';
+
+jest.mock('fs');
+
+class TestArtifactProvider extends BaseArtifactProvider {
+  protected doDownloadArtifact(
+    _artifact: RemoteArtifact,
+    _downloadDirectory: string
+  ): Promise<string> {
+    throw new Error('Method not implemented.');
+  }
+
+  protected doListArtifactsForRevision(
+    _revision: string
+  ): Promise<RemoteArtifact[]> {
+    throw new Error('Method not implemented.');
+  }
+}
+
+/** Returns a new AwsLambdaTarget test instance. */
+function getAwsLambdaTarget(): AwsLambdaTarget {
+  return new AwsLambdaTarget(
+    {
+      ['testKey']: 'testValue',
+    },
+    new TestArtifactProvider({
+      repoName: 'testName',
+      repoOwner: 'testRepo',
+      ['testKey']: 'testValue',
+    })
+  );
+}
+
+function setAwsEnvironmentVariables() {
+  process.env.AWS_ACCESS_KEY_ID = 'test aws access key';
+  process.env.AWS_SECRET_ACCESS_KEY = 'test aws secret access key';
+}
+
+describe('get layer name environment variable', () => {
+  test('default environment variable', () => {
+    if ('AWS_LAYER_NAME' in process.env) {
+      delete process.env.AWS_LAYER_NAME;
+    }
```

Comment:
Perhaps would be a bit safer to clear the variable in a `beforeEach` block, for the future.

Reply:
Good catch, the layer name should come in as project-specific config, not environment variable. These lines should go :)